build.sbt

```diff
@@ -133,6 +135,7 @@ lazy val core = (project in file("core"))
     libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value,
     libraryDependencies += "com.github.scopt" %% "scopt" % "3.7.1",
     libraryDependencies += "com.lihaoyi" %% "sourcecode" % "0.2.7",
+    libraryDependencies += scalatest,
```

Comment:
I am very sensitive on the dependency of the libraries recently.
In my opinion, the SpinalSpec is not for core, it's for sim.
the core mainly contains facilities that elaborate logic, so my favorite fundamental component Stream have been put to the lib.
So I the sim is focused on simulation where Spec can be layed out.


Btw, the existing Tester are all located in tester directory?
How about keep that consistent?

Reply:
IIUC `sim` is for internals, not intended to be imported at all by the user. Users import `core.sim._`, which is the place where I put `SpinalSpec` here.

Btw I prefer how @andreasWallner added this dependency, so I suggest to merge his PR https://github.com/SpinalHDL/SpinalHDL/pull/970 first, then I solve conflicts by just taking his `build.sbt`.

About where the tests are, it is more compliant with `sbt` structure (@andreasWallner [agrees with it](https://github.com/SpinalHDL/SpinalHDL/pull/970)) so that if I'm seeking tests for `SpinalSpec.scala`, I can easily find them. It could be moved to `tester` but it would be more messy to mix all tests without any structure. I think `tester` is good for integration tests, but here it is more local, it is unit testing.

Reply:
Good point, #970 isn't about sim so why does it rearrange the build libraries?

Reply:
Maybe scala-test should have been integrated to scala itself as it is a very very much used library, so is it a big issue to have it as a depedency of SpinalHDL ? @Readon

Reply:
I saw some one is using uTest framework instead. 
IMO, If scalatest dependency is not propagated in published version, then it could be good.

Reply:
I need it to provide the feature: https://github.com/SpinalHDL/SpinalHDL/pull/968/files#diff-8958280689950ea4b1f86c34cbcfdc12116d668559bfc108dbd358e70785c6dfR3-R4

Reply:
I have noticed that.
emm, if we agreed on instist using scalatest but others, it can be resonable to import that dependency.

Reply:
I want to help users writing tests for hardware without having them to deal with `scalatest`.

Do you think we should base that on another tool than `scalatest`? Or if we should base that on nothing and interact directly with lower-level testing stuff?

Reply:
@kleinai I had added scalatest to be able to be able to add test to the core library. I had expected that sbt would support test dependencies, but it that is not the case, e.g. looking at this: https://www.scala-sbt.org/1.x/docs/Testing.html#Basics

Reply:
@kleinai I just understood that I could limit the scope of the dependency to tests only via adding `% test` - seen that before, not sure why I forgot. Will do that change in my PR, which means it's still a topic here.

Reply:
Ah so for this reason maybe the approach here with a `val` is better because it makes it possible to add `% test` to all projects other than `core`?

But as I need it as normal dependency for this PR, maybe it is okay too to make it as a normal dependency everywhere?

Reply:
@numero-744 there is still no reason to add it as a regular dependency to core though?

Reply:
This way the user would have to add `import spinal.lib._` to use it?

Reply:
@numero-744 yes, just like with `SpinalFormalFunSuite`

Reply:
or let's move SpinalSpec and SpinalFormalFunSuite to the spinal tester module ? and add that module into the releases (after some cleaning of the tester main module XD)

Reply:
@Dolu1990 that came to mind yesterday as well, might make sense if we want to skip the dependency on scalatest - I don't think there are many SpinalHDL users that use `core` but not `lib`

Reply:
See PR : https://github.com/SpinalHDL/SpinalHDL/pull/980 which implement it.

Reply:
According to #980, I'll rebase and move my stuff to `tester` dependency without moving it from `core.sim` package once @andreasWallner validates the suggested updated API.